pallets/parachain-staking/src/migrations.rs

```diff
@@ -17,19 +17,113 @@
 //! # Migrations
 use crate::{
 	pallet::{migrate_nominator_to_delegator_state, RoundIndex},
-	BalanceOf, CandidateState, CollatorCandidate, CollatorState2, Config, DelegatorState,
-	ExitQueue2, NominatorState2, Points, Round, Staked,
+	BalanceOf, Bond, CandidateState, CollatorCandidate, CollatorState2, Config, DelegatorState,
+	ExitQueue2, NominatorState2, Pallet, Points, Round, Staked,
 };
 #[cfg(feature = "try-runtime")]
 use crate::{Collator2, Delegator, Nominator2};
 #[cfg(feature = "try-runtime")]
+use frame_support::traits::OnRuntimeUpgradeHelpersExt;
+#[cfg(feature = "try-runtime")]
 use frame_support::Twox64Concat;
+extern crate alloc;
+#[cfg(feature = "try-runtime")]
+use alloc::format;
 use frame_support::{
 	pallet_prelude::PhantomData,
 	traits::{Get, OnRuntimeUpgrade},
 	weights::Weight,
 };
-use sp_std::collections::btree_map::BTreeMap;
+use sp_std::{collections::btree_map::BTreeMap, vec::Vec};
+
+/// Migration to properly increase maximum delegations per collator
+/// This migration can be used to recompute the top and bottom delegations whenever
+/// MaxDelegatorsPerCandidate changes (works for decrease as well)
+pub struct IncreaseMaxDelegationsPerCandidate<T>(PhantomData<T>);
+impl<T: Config> OnRuntimeUpgrade for IncreaseMaxDelegationsPerCandidate<T> {
+	fn on_runtime_upgrade() -> Weight {
+		let (mut reads, mut writes) = (0u64, 0u64);
+		for (account, state) in <CandidateState<T>>::iter() {
+			reads += 1u64;
+			// 1. collect all delegations into single vec and order them
+			let mut all_delegations = state.top_delegations.clone();
+			let mut starting_bottom_delegations = state.bottom_delegations.clone();
+			all_delegations.append(&mut starting_bottom_delegations);
+			// sort all delegations from greatest to least
+			all_delegations.sort_unstable_by(|a, b| b.amount.cmp(&a.amount));
```

Comment:
Note: this is `O(n * log(n))`, which could become significant with larger sets. In the future, it might be good to attempt to estimate weight more precisely here.

Reply:
...although this list is already sorted, right? We should make to use a sort algo that is efficient on sorted lists.

Reply:
Should make a test benchmark to see how much weight sorting a list of size N elements takes (when it starts sorted, unsorted)

Reply:
> ...although this list is already sorted, right? We should make to use a sort algo that is efficient on sorted lists.

It is not sorted. The top is greatest to least, bottom is least to greatest so appending bottom to top does not imply the result is sorted greatest to least (because of bottoms opposite ordering)...We could reverse the bottom and then append it to the top which would be more efficient. I just don't know if its necessary.

Reply:
Reversing the list is a good idea, esp. if we need this migration again